node/cn/backend.go

```diff
@@ -259,28 +259,14 @@ func New(ctx *node.ServiceContext, config *Config) (*CN, error) {
 		return nil, err
 	}
 
-	// Set AcceptTxs flag in 1CN case to receive tx propagation.
-	if chainConfig.Istanbul != nil {
-		istanbulExtra, err := types.ExtractIstanbulExtra(cn.blockchain.Genesis().Header())
-		if err != nil {
-			logger.Error("Failed to decode IstanbulExtra", "err", err)
-		} else {
-			if len(istanbulExtra.Validators) == 1 {
-				atomic.StoreUint32(&cn.protocolManager.acceptTxs, 1)
-			}
-		}
+	if err := cn.setAcceptTxs(); err != nil {
+		logger.Error("Failed to decode IstanbulExtra", "err", err)
```

Comment:
Wouldn't it be better to do this in a function that prints an error? Do it need to get an error and print it back out?

Reply:
You're right. However, if we want to test a function, we want to know whether it returns an error when expected.